cmake/std/atdm/common/toss3/environment_tlcc2.sh

```diff
@@ -47,7 +47,7 @@ if [ "$ATDM_CONFIG_COMPILER" == "INTEL" ]; then
     module load sems-boost/1.66.0/base
 
     module swap mkl/18.0.0.128 mkl/18.0.5.274
-    module load gcc/4.9.3
+    module load cde/compiler/gcc/7.2.0
```

Comment:
Wow, that easy?

Reply:
From my experience, yes it is that easy.

Reply:
@prwolfe, and @etphipp,

What impact, if any, would this have on the existing upstream TPLs that may have been built with GCC 4.9.3?  I guess if they don't use the C++ standard library, then it should be okay?  What about the standard C library (that TPLs like SuperLUDist must be using at least some)  with these two GCC versions.  Otherwise, are the header files and standard library in GCC 7.2.0 guaranteed to be backward compatible with the header files and standard library with GCC 4.9.3?

That is my main concern here.  I think we will need to build and run the full Trilinos test suite to be sure.  And someone really should be testing against SPARC and EMPIRE.  (That is what the ATDM DevOps team has done for 2+ years.)

Reply:
That certainly can be an issue and this likely wouldn't work in all cases.  But to me the question is does it work in our case or not, which can easily be tested by just trying to compile and link the codes.  The ideal solution would be to recompile everything with gcc7 (or any gcc version where the c++14 support is not experimental), and that should be done long term even if swapping the compiler works in this case.

Reply:
We could be more conservative and swap for GCC 5.4 or so if that exists. But generally GCC versions are supposed to ABI compatible ...